### ISSUE TITLE
Issues in Vietnamese tokenizer

### ISSUE THREAD

**Issue by alejandrojcastaneira** [explosion/spaCy#8097]
## How to reproduce the behaviour

Hello,

I have noticed two issues in Vietnamese tokenization.

Example.

````
import spacy

nlp= spacy.blank("vi")


doc = nlp("""Đây là một văn bản bằng tiếng Việt
            Sau đó, đây là một văn bản khác bằng ngôn ngữ này""")

print(doc.text)
````

Produces:

`Đây là một văn_bản bằng tiếng Việt Sau đó, đây là một văn_bản khác bằng ngôn_ngữ này`

1 - New lines are being stripped

2 - Spaces are being arbitrarily replaced with the  "_" character

Best regards

## Your Environment

* Operating System: Ubuntu 20.04.1 LTS
* Python Version Used: 3.6.9
* spaCy Version Used: 2.3.4

**Comment by adrianeboyd** [comment]
Thanks for the report! This seems to be hard-coded behavior coming from the third-party library `pyvi`. We might be able to provide a workaround so that the original text is not modified, but I'd have to look into the details.

Do you know of any other libraries for Vietnamese tokenization that preserve whitespace correctly?

**Comment by alejandrojcastaneira** [comment]
thank you for the quick reply! unfortunately, I don't know about other libraries which I could recommend, but after researching a little bit I found out that some tokenizer libraries in Vietnamese report to marks multi-syllable tokens with underscores instead of spaces. So probably this could be the expected behaviour, if this the case, maybe we should consider the option to control this?

The stripping of the new lines definitely looks like a pyvi issue, perhaps, I should report this issue there?

**Comment by adrianeboyd** [comment]
I suspect that's the intended behavior for the tokenizer. We can add a bit of a workaround to preserve the whitespace as we want to for `spacy`: https://github.com/explosion/spaCy/pull/8099

Since this changes the saved model format for Vietnamese models, we would wait until v3.1.0 to incorporate this change.

**Comment by adrianeboyd** [comment]
Okay, let's hope this will work better after #8099. It will be released as part of v3.1.0. (For some reason it didn't get closed automatically when it was merged, github has been flaky about this recently.)
